mp/src/game/client/momentum/ui/leaderboards/ClientTimesDisplay.cpp

```diff
@@ -102,6 +103,8 @@ void CClientTimesDisplay::OnThink()
             GetClientVoiceMgr()->StopSquelchMode();
         }
     }
+
+    if (input()->IsKeyDown(KEY_ESCAPE)) { Close(); }
```

Comment:
The way that the ChatPanel does it is listening for when escape is pressed is actually on the `OnKeyCodeReleased(KeyCode code)` override, as seen here:
https://github.com/momentum-mod/game/blob/develop/mp/src/game/client/momentum/ui/chat/ChatPanel.h#L143

And implementation here:
https://github.com/momentum-mod/game/blob/develop/mp/src/game/client/momentum/ui/chat/ChatPanel.cpp#L199

We should definitely do the same here inside CClientTimesDisplay, and you can remove the `vgui/IInput` include as well.

Reply:
I also notice that this panel doesn't have keyboard input so this override may not be called. You'll then want to `SetKeyboardInputEnabled(true);` when the panel is toggled open, and have it set to false in the constructor, and also set it to false where you re-allow escape to show.

Reply:
Thanks, I tried that initially but I did not know about the `SetKeyboardInputEnabled` function at the time. I will commit again with the requested changes.